pkg/controller/common/reconciler.go

```diff
@@ -214,26 +213,37 @@ func (r *BaseIngressReconciler) reconcileDeletion(ctx context.Context, ci networ
 	// If particular namespace has only one ingress object then after deletion namespace will have empty ingress object
 	// So remove namespace from SMMR
 	if len(ingressList.Items) == 1 {
-		if err := r.Client.Get(ctx, types.NamespacedName{Name: smmrName, Namespace: smmrNamespace}, smmr); err != nil {
-			return err
-		}
-		for i, val := range smmr.Spec.Members {
-			if val == ci.GetNamespace() {
-				smmr.Spec.Members = append(smmr.Spec.Members[:i], smmr.Spec.Members[i+1:]...)
-				break
+		for i := range ingressList.Items {
+			// In order to double check that we are reconciling proper ingress check with name and namespace
+			if ci.GetNamespace() == ingressList.Items[i].Namespace && ci.GetName() == ingressList.Items[i].Name {
+				smmr := &maistrav1.ServiceMeshMemberRoll{}
+				if err := r.Client.Get(ctx, types.NamespacedName{Name: smmrName, Namespace: smmrNamespace}, smmr); err != nil {
+					return err
+				}
+				for i, val := range smmr.Spec.Members {
+					if val == ci.GetNamespace() {
+						smmr.Spec.Members = append(smmr.Spec.Members[:i], smmr.Spec.Members[i+1:]...)
+						break
+					}
+				}
+				if err := r.Client.Update(ctx, smmr); err != nil {
+					return err
+				}
 			}
 		}
-		if err := r.Client.Update(ctx, smmr); err != nil {
-			return err
+	}
+	for i := range ci.GetFinalizers() {
+		// remove finalyzer which was added by knative-openshift-ingress
+		if ci.GetFinalizers()[i] == "ocp-ingress" {
+			ci.SetFinalizers(ci.GetFinalizers()[i+1:])
+			break
 		}
 	}
-	// remove finalyzer in order to remove the ingress object
-	ci.SetFinalizers(ci.GetFinalizers()[1:])
 	return r.Client.Update(ctx, ci)
 }
 
-// appendIfAbsent append namespace to member if its not exist
-func appendIfAbsent(members []string, routeNamespace string) ([]string, bool) {
+// AppendIfAbsent append namespace to member if its not exist
+func (r *BaseIngressReconciler) AppendIfAbsent(members []string, routeNamespace string) ([]string, bool) {
```

Comment:
Why add this to the Reconciler? We just need to export it by giving it the uppercase name.